src/Symfony/Component/HttpKernel/EventListener/LocaleListener.php

```diff
@@ -68,10 +68,15 @@ private function setLocale(Request $request)
     {
         if ($locale = $request->attributes->get('_locale')) {
             $request->setLocale($locale);
-        } elseif ($this->useAcceptLanguageHeader && $this->enabledLocales && ($preferredLanguage = $request->getPreferredLanguage($this->enabledLocales))) {
+        } elseif (!$this->useAcceptLanguageHeader) {
+            return;
```

Comment:
I prefered the nested version. The current one is confusion because setLocale is called twice

Reply:
Oh right :+1: